zerver/webhooks/teamcity/view.py

```diff
@@ -42,6 +42,8 @@ def api_teamcity_webhook(request: HttpRequest, user_profile: UserProfile,
     message = payload['build']
 
     build_name = message['buildFullName']
+    if 'branchDisplayName' in message:
```

Comment:
Your tests are failing because this should be `if message['build']['branchDisplayName']:` I think! :)

Reply:
I'm not sure of the best way to write that condition, but it was not the reason for failed tests.
It was the expected subject